src/prefect/futures.py

```diff
@@ -455,14 +470,14 @@ def _collect_futures(futures, expr, context):
         return expr
 
     # Get final states for each future
-    states = []
+    states: list[State] = []
```

Comment:
State is generic, so this is missing a parameter.

```suggestion
    states: list[State[Any]] = []
```